src/rgw/rgw_lua.cc

```diff
@@ -134,14 +145,38 @@ int add_package(const DoutPrefixProvider *dpp, rgw::sal::Store* store, optional_
 
 int remove_package(const DoutPrefixProvider *dpp, rgw::sal::Store* store, optional_yield y, const std::string& package_name) {
   librados::ObjectWriteOperation op;
-  op.omap_rm_keys(std::set<std::string>({package_name}));
-  const auto ret = rgw_rados_operate(dpp, *(static_cast<rgw::sal::RadosStore*>(store)->getRados()->get_lc_pool_ctx()),
-    PACKAGE_LIST_OBJECT_NAME, &op, y);
-
-  if (ret < 0) {
+  size_t pos = package_name.find(" ");
+  //remove specfic version of the the package
```

Comment:
why do you need to remove the specific versions as well as all the other versions?
why not just remove all versions of the package in this case - since you do that in line 159 and on?

Reply:
> why do you need to remove the specific versions as well as all the other versions?
> why not just remove all versions of the package in this case - since you do that in line 159 and on?

trying to match `luarocks` behavior, if a version was provided in the removal - only the specific one will be removed. otherwise, all the versions will be removed.
the case where a specific version is provided is between lines 151-157 (ending with `return 0` in order to avoid removing all the other versions).
the latter case is in line 159 and on, removing all versions since no version specification was provided.